examples/webgl_buffergeometry.html

```diff
@@ -51,105 +58,133 @@
+					const pC = new THREE.Vector3();
 
-				const cb = new THREE.Vector3();
-				const ab = new THREE.Vector3();
+					const cb = new THREE.Vector3();
+					const ab = new THREE.Vector3();
 
-				for ( let i = 0; i < triangles; i ++ ) {
+					for ( let i = 0; i < triangles; i ++ ) {
 
-					// positions
+						// positions
 
-					const x = Math.random() * n - n2;
-					const y = Math.random() * n - n2;
-					const z = Math.random() * n - n2;
+						const x = Math.random() * n - n2;
+						const y = Math.random() * n - n2;
+						const z = Math.random() * n - n2;
 
-					const ax = x + Math.random() * d - d2;
-					const ay = y + Math.random() * d - d2;
-					const az = z + Math.random() * d - d2;
+						const ax = x + Math.random() * d - d2;
+						const ay = y + Math.random() * d - d2;
+						const az = z + Math.random() * d - d2;
 
-					const bx = x + Math.random() * d - d2;
-					const by = y + Math.random() * d - d2;
-					const bz = z + Math.random() * d - d2;
+						const bx = x + Math.random() * d - d2;
+						const by = y + Math.random() * d - d2;
+						const bz = z + Math.random() * d - d2;
 
-					const cx = x + Math.random() * d - d2;
-					const cy = y + Math.random() * d - d2;
-					const cz = z + Math.random() * d - d2;
+						const cx = x + Math.random() * d - d2;
+						const cy = y + Math.random() * d - d2;
+						const cz = z + Math.random() * d - d2;
 
-					positions.push( ax, ay, az );
-					positions.push( bx, by, bz );
-					positions.push( cx, cy, cz );
+						positions.push( ax, ay, az );
+						positions.push( bx, by, bz );
+						positions.push( cx, cy, cz );
 
-					// flat face normals
+						// flat face normals
 
-					pA.set( ax, ay, az );
-					pB.set( bx, by, bz );
-					pC.set( cx, cy, cz );
+						pA.set( ax, ay, az );
+						pB.set( bx, by, bz );
+						pC.set( cx, cy, cz );
 
-					cb.subVectors( pC, pB );
-					ab.subVectors( pA, pB );
-					cb.cross( ab );
+						cb.subVectors( pC, pB );
+						ab.subVectors( pA, pB );
+						cb.cross( ab );
 
-					cb.normalize();
+						cb.normalize();
 
-					const nx = cb.x;
-					const ny = cb.y;
-					const nz = cb.z;
+						const nx = cb.x;
+						const ny = cb.y;
+						const nz = cb.z;
 
-					normals.push( nx, ny, nz );
-					normals.push( nx, ny, nz );
-					normals.push( nx, ny, nz );
+						normals.push( nx, ny, nz );
+						normals.push( nx, ny, nz );
+						normals.push( nx, ny, nz );
 
-					// colors
+						// colors
 
-					const vx = ( x / n ) + 0.5;
-					const vy = ( y / n ) + 0.5;
-					const vz = ( z / n ) + 0.5;
+						const vx = ( x / n ) + 0.5;
+						const vy = ( y / n ) + 0.5;
+						const vz = ( z / n ) + 0.5;
 
-					color.setRGB( vx, vy, vz );
+						color.setRGB( vx, vy, vz );
 
-					colors.push( color.r, color.g, color.b );
-					colors.push( color.r, color.g, color.b );
-					colors.push( color.r, color.g, color.b );
+						if ( data.VertexAlpha ) {
 
-				}
+							colors.push( color.r, color.g, color.b, data.alpha );
```

Comment:
Would not the demo be more interesting if you apply a different alpha value per face? Otherwise I wonder why the code does not just use `Material.opacity`?

Reply:
How about a random alpha value for each of the triangles? Like the rgb colors.

Reply:
Sounds good!